### DISCUSSION TITLE
Could I reference a color from DaisyUI theme in `theme` section of my `tailwind.config.js`

### DISCUSSION THREAD

**Discussion by dominikkrulak** [saadeghi/daisyui#1987] [Q&A] [answered]
I use the official Tailwind CSS [typography plugin](https://tailwindcss.com/docs/typography-plugin) and there are some default styles I'd like to change or write a new ones. I wouldn't want to use Tailwind's colors but rather DaisyUI theme's colors.

DaisyUI themes `light` and `dark` have colors overwritten.

Could It be done?

**tailwind.config.js**
```
//...
theme: {
  extend: {
    typography: (theme) => ({
      code: {
        // Rather use some DaisyUI color
        backgroundColor: theme('colors.neutral.300'),
      },
    }),
  },
},
plugins: [require('@tailwindcss/typography'), require('daisyui')],
daisyui: {
  themes: [
    {
      light: {
        "primary": "#111111",
        "secondary": "#FF5A00",
        "accent": "#1BE480",
        "neutral": "#C0C0C0",
        "base-100": "#ffffff",
        "info": "#F5F5F5",
        "success": "#16a34a",
        "warning": "#FFE5B4",
        "error": "#FF2400",
        "--rounded-btn": ".75rem",
        "--btn-text-case": "none",
        "--border-btn": "3px",
        // Transform scale property causes bug
        // for links with inner element with "absolute inset-0"
        "--btn-focus-scale": "none",
      },
      dark: {
        "primary": "#FFFFFF",
        "secondary": "#FF8949",
        "accent": "#39FF14",
        "neutral": "#464646",
        "base-100": "#353535",
        "base-200": "#262626",
        "base-300": "#1B1B1B",
        "info": "#FBEC5D",
        "success": "#355E3B",
        "warning": "#FF9966",
        "error": "#E60026",
        "--rounded-btn": ".75rem",
        "--btn-text-case": "none",
        "--border-btn": "3px",
        // Transform scale property causes bug
        // for links with inner element with "absolute inset-0"
        "--btn-focus-scale": "none",
      },
    },
  ],
}
```

**Comment by dominikkrulak** [comment, marked as answer]
This already have the [answer](https://github.com/saadeghi/daisyui/discussions/415#discussioncomment-1995833).

**Reply by saadeghi** [reply]
Yes. a few more things to add:  
- `theme()` function is for Tailwind's [theme configuration](https://tailwindcss.com/docs/theme) where you can define all the values or use a preset. Not to be confused with daisyUI themes.
- daisyUI [overrides the default color CSS variables](https://github.com/saadeghi/daisyui/blob/master/src/components/styled/typography.css) of `@tailwindcss/typography` plugin with [`base-content` color](https://daisyui.com/docs/colors/) so any element inside `prose` works with all daisyUI themes by default. So I would suggest to [edit the CSS variables](https://tailwindcss.com/docs/typography-plugin#adding-custom-color-themes) values if you want to edit those colors. Like I did [here](https://github.com/saadeghi/daisyui/blob/master/src/components/styled/typography.css) (use the table from [this page](https://daisyui.com/docs/colors/) as a refrence)
For example `'--tw-prose-body': hsl(var(--p)),` sets `primary` color for `prose`)

**Reply by dominikkrulak** [reply]
So I ended up with following:
```
const daisyuiColors = require('daisyui/src/theming/themes');
//...
theme: {
  extend: {
    typography: (theme) => ({
      '*[data-theme="light"] code': {
        backgroundColor: daisyuiColors['[data-theme=light]']['base-200'],
      },
      '*[data-theme="dark"] code': {
        backgroundColor: daisyuiColors['[data-theme=dark]']['base-200'],
      },
    }),
  },
},
```

**Reply by dominikkrulak** [reply]
@saadeghi I've noticed that smart override of all `prose` elements by daisyUI themes colors. But there are examples where you must extend styles if defaults don't cover it up like the example above.

**Reply by dominikkrulak** [reply]
> Yes. a few more things to add:
> 
> * `theme()` function is for Tailwind's [theme configuration](https://tailwindcss.com/docs/theme) where you can define all the values or use a preset. Not to be confused with daisyUI themes.
> * daisyUI [overrides the default color CSS variables](https://github.com/saadeghi/daisyui/blob/master/src/components/styled/typography.css) of `@tailwindcss/typography` plugin with [`base-content` color](https://daisyui.com/docs/colors/) so any element inside `prose` works with all daisyUI themes by default. So I would suggest to [edit the CSS variables](https://tailwindcss.com/docs/typography-plugin#adding-custom-color-themes) values if you want to edit those colors. Like I did [here](https://github.com/saadeghi/daisyui/blob/master/src/components/styled/typography.css) (use the table from [this page](https://daisyui.com/docs/colors/) as a refrence)
>   For example `'--tw-prose-body': hsl(var(--p)),` sets `primary` color for `prose`)

Of course there could be more ways to do it as you hinted but for now I stick to [this one](https://github.com/saadeghi/daisyui/discussions/1987#discussioncomment-6153011).

Thank you @saadeghi
